crates/bevy_render/src/mesh/components.rs

```diff
@@ -96,3 +116,10 @@ impl From<&Mesh3d> for AssetId<Mesh> {
         mesh.id()
     }
 }
+
+// The `check_visibility` system needs to mark 3D meshes as visible.
+fn add_mesh3d_visibility_class(mut world: DeferredWorld<'_>, entity: Entity, _: ComponentId) {
+    if let Some(mut visibility_class) = world.get_mut::<VisibilityClass>(entity) {
+        visibility_class.push(TypeId::of::<Mesh3d>());
+    }
+}
```

Comment:
These hooks could be generic one generic function and live in the visibility module for slightly less boilerplate? Unless that doesn't work with the macro.

Reply:
I like this suggestion. Should work fine.

Reply:
Done.